content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +168,64 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
+
+For every `dvc.yaml` file, a matching `dvc.lock` (YAML) file usually exists.
+It's created or updated by DVC commands such as `dvc run` and `dvc repro`.
+`dvc.lock` describes the latest pipeline state. It has several purposes:
+
+- Tracking of intermediate and final results of a pipeline — similar to
+  [`.dvc` files](#dvc-files). This is done with a mapping of file and directory
+  paths in the <abbr>project</abbr> to their locations (hash values) in the
+  <abbr>cache</abbr> or [remote storage](/doc/command-reference/remote).
+- Allow DVC to detect when stage definitions, or their dependencies have
+  changed. Such conditions invalidate states, requiring their reproduction (see
+  `dvc status`, `dvc repro`).
+- `dvc.lock` is needed internally for several DVC commands to operate, such as
+  `dvc checkout`, `dvc get`, and `dvc import`.
+
+Here's an example `dvc.lock` based on the one in [`dvc.yaml`](#dvcyaml-file)
+above:
+
+```yaml
+stages:
+  features:
+    cmd: jupyter nbconvert --execute featurize.ipynb
+    deps:
+      - path: data/clean
+        md5: d8b874c5fa18c32b2d67f73606a1be60
+    params:
+      params.yaml:
+        levels.no: 5
+    outs:
+      - path: features
+        md5: 2119f7661d49546288b73b5730d76485
+      - path: performance.json
+        md5: ea46c1139d771bfeba7942d1fbb5981e
+      - path: logs.csv
+        md5: f99aac37e383b422adc76f5f1fb45004
+```
+
+Stage commands are listed again in `dvc.lock`, in order to know when their
+definitions change in the `dvc.yaml` file.
+
+Regular <abbr>dependencies</abbr> and all kinds of <abbr>outputs</abbr>
+(including [metrics](/doc/command-reference/metrics) and
+[plots](/doc/command-reference/plots) files) are also listed (per stage) in
+`dvc.lock`, but with an additional field to store the hash value of each file or
+directory tracked by DVC:
```

Comment:
...on the other hand, if we keep the explanation about mapping to the cache/storage, should we mention it here again or even provide a small example of the corresponding .dvc/cache/ `tree` ?

Reply:
This no longer applies per https://github.com/iterative/dvc.org/pull/1448#discussion_r445116497